Documentation improvements for the validator migration (#48)
onmax/nimiq-developer-center: migration/migration-validators.md

- Keys and corresponding address generation
- Validator registration via transactions

### Prerequisites

> 1. Make sure you have [Node.js](https://nodejs.org) >= v16.20.2 installed.

Reviewer: Node V18 is the latest maintained version. We shouldn't endorse any unmaintained versions: https://nodejs.org/en/about/previous-releases

### Step 3: Run the validator registration tool

To register your validator and the corresponding keys on the PoW Mainnet, you must specify the arguments as follows:

```shell
node validator-registration.js --validator validator-keys.json --network main
```

> The tool will connect to the Nimiq PoW Mainnet chain. Upon consensus, the tool creates and signs the validator registration transactions and send them to the network, effectively registering your validator. Note that these transactions are a bundle of 6, and each one has a nominal fee of 1 Luna. Please ensure you have funded your validator address &mdash; as instructed in the previous step &mdash; as the transaction fees are paid from this address.

Reviewer 2: `sends` was right, as it relates to the tool.

To execute the tool successfully, you need to ensure that you are fully synced and in consensus **within the PoW chain**. This requires starting the PoW client with an RPC server enabled on your imported validator address (this might take a while). As in [this example](https://github.com/nimiq/core-js/blob/master/clients/nodejs/sample.conf#L163), you can do it in your configuration file by adding your address and key pair.

Alternatively, you can start the RPC server by cloning the [Nimiq CoreJS repository](https://github.com/nimiq/core-js?tab=readme-ov-file#quickstart) then following the quickstart instructions and finally running the following command:

```shell
node clients/nodejs/index.js --dumb --network=main --rpc=8648 --wallet-seed=private_keypublic_key
```

Note that `private_keypublic_key` is the concatenation of your private key and public key in that order.

Before executing the activation tool, make sure you run ```cargo build --release``` within the PoS repository to compile the code.

> Once you are in consensus in the PoW chain, proceed to execute the migration tool by running the following command **in the PoS chain client**, including the path to the configuration file containing your validator data and specifying the RPC server to be used:

Reviewer 2: ```suggestion
Once you are in consensus in the PoW chain, proceed to execute the migration tool by running the following command **in the PoS chain client** directory, including the path to the configuration file containing your validator data and specifying the RPC server to be used:
```

> Once you are in consensus in the PoW chain, proceed to execute the migration tool by running the following command **in the PoS chain client** directory, including the path to the configuration file containing your validator data and specifying the RPC server to be used:

Reviewer: ```suggestion
Once you have consensus on the PoW chain, proceed to execute the migration tool by running the following command **in the PoS chain client** directory, including the path to the configuration file containing your validator data and specifying the RPC server to be used:
```

```shell
cargo run --release --bin nimiq-pow-migration --url "url-according-your-configuration" --config client.toml
```

> Where `url-according-your-configuration` is the PoW client RPC url. For instance: `http://172.0.0.1:8648`.

Reviewer 2: ```suggestion
Where `url-according-your-configuration` is the PoW client RPC url. For example: `http://172.0.0.1:8648`.
```
